media/js/base/core-datalayer-events.es6.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+function getDownloadDetails(downloadHref) {
+    const downloadURL = new URL(downloadHref);
+    const productParam = downloadURL.searchParams.get('product');
+    const productSplit = productParam.split('-');
+    const product = productSplit[0];
+    const releaseChannel =
+        productSplit[1] === 'latest' ? 'release' : productSplit[1];
+    const platform = downloadURL.searchParams.get('os');
+    const lang = downloadURL.searchParams.get('lang');
+
+    const eventObj = { event: 'product_download' };
+    eventObj['product'] = product;
+    eventObj['platform'] = platform;
+    eventObj['release_channel'] = releaseChannel;
+    eventObj['download_language'] = lang;
+
+    return eventObj;
+}
+
+function pushEvent(downloadDetails) {
+    //send event
+    window.dataLayer.push(downloadDetails);
+}
+
+const init = () => {
+    // add watcher to all links to download.mozilla.org
+    const downloadLinks = document.querySelectorAll(
+        'a[href^="https://download.mozilla.org/"]'
+    );
+    for (const downloadLink of downloadLinks) {
```

Comment:
I think Babel might handle `for ... of` OK in older browsers, but definitely worth checking as it's not something we use a lot I think (especially as you're passing a NodeList and not a normal array).